### PR TITLE
SONARGO-35 Support Go 1.23

### PR DESCRIPTION
**Pull request by petertrr**
[SONARGO-35](https://sonarsource.atlassian.net/browse/SONARGO-35)

[SONARGO-35]: https://sonarsource.atlassian.net/browse/SONARGO-35?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: sonar-go-to-slang/goparser_test.go [hunk 1]
```diff
@@ -88,10 +87,7 @@ func Test_all_go_files(t *testing.T) {
 			panic(err2)
 		}
 
-		if !reflect.DeepEqual(jsonExpected, jsonActual) {
-			fmt.Printf("Failed to match expected results for file: %#v\n", file)
-			t.Fatalf("got: %#v\nexpected: %#v", jsonActual, jsonExpected)
-		}
+		assert.Equal(t, string(expectedData), actual, "Failed to match expected results for file: %#v\n", file)
```
Comment (petertrr) [inline sonar-go-to-slang/goparser_test.go:90]: `testify` uses `DeepEqual` under the hood, and also produces a readable diff

File: sonar-go-plugin/build.gradle [hunk 2]
```diff
@@ -86,7 +86,7 @@ shadowJar {
         exclude 'tmp/**'
     }
     doLast {
-        enforceJarSizeAndCheckContent(shadowJar.archiveFile.get().asFile, 8_000_000L, 8_500_000L)
+        enforceJarSizeAndCheckContent(shadowJar.archiveFile.get().asFile, 9_000_000L, 9_500_000L)
```
Comment (petertrr) [inline sonar-go-plugin/build.gradle:89]: Apparently, newer Go version produce larger binaries :shrug:

File: sonar-go-to-slang/go.mod [hunk 3]
```diff
@@ -1,3 +1,11 @@
 module github.com/SonarSource/slang/sonar-go-to-slang
 
-go 1.21
+go 1.23.4
```
Comment (petertrr) [inline sonar-go-to-slang/go.mod:3]: As new `range` options are syntactically the same and we don't have any advanced analysis in the Go analyzer, no changes in the parser and/or converter are needed

File: sonar-go-to-slang/resources/ast/loops.go.source [hunk 4]
```diff
@@ -43,4 +48,15 @@ OUTER:
...
+	// range over integer; requires Go 1.22+
+	for i := range 10 {
+        fmt.Println(10 - i)
+    }
+
+    // range over an iterator, requires Go 1.23+
+    l := []int{1, 2, 3}
+    for v := range slices.Values(l) {
+        fmt.Println(v)
+    }
```
Comment (mstachniuk) [inline sonar-go-to-slang/resources/ast/loops.go.source:61]: Maybe more examples should be added for range over an iterator?

**Review by mstachniuk — APPROVED** [review]
LGTM

**Pull request merged**